server/frontend/src/components/ProviderKey.vue

```diff
@@ -0,0 +1,73 @@
+<template>
+  <div class="mt-strong">
+    <strong>{{ providerHostname }}:</strong>
+    <div class="form-inline mt-light">
+      <div class="input-group">
+        <input
+          type="text"
+          class="form-control"
+          placeholder="API key..."
+          v-model="key"
+        />
+        <span class="input-group-btn">
+          <button type="button" class="btn btn-success" v-on:click="saveKey">
+            Save
+          </button>
+        </span>
+      </div>
+      <button
+        type="button"
+        class="btn btn-danger"
+        v-on:click="removeKey"
+        title="Remove key"
+      >
+        <span class="glyphicon glyphicon-trash" aria-hidden="true"></span>
+      </button>
+    </div>
+  </div>
+</template>
+
+<script>
+export default {
+  props: {
+    providerId: {
+      type: Number,
+      required: true,
+    },
+    providerHostname: {
+      type: String,
+      required: true,
+    },
+  },
+  data: () => ({
+    key: null,
+  }),
+  mounted() {
+    const storedKey = localStorage.getItem(this.localStorageKey);
+    if (storedKey) this.key = storedKey;
+  },
+  computed: {
+    localStorageKey() {
+      return "provider-" + this.providerId;
+    },
+  },
+  methods: {
+    saveKey() {
+      localStorage.setItem(this.localStorageKey, this.key);
```

Comment:
It would be pretty nice to make an authenticated query towards the Bugzilla server (as you already have the hostname here) to check if the api key is valid.

You just need to make a query on [whoami](https://bmo.readthedocs.io/en/latest/api/core/v1/user.html#who-am-i) and display the real name of the user (or an error)